libtransact/src/context/manager/thread/mod.rs

```diff
@@ -138,18 +138,20 @@ impl ContextManagerThread {
                     break;
                 }
                 Err(err) => {
-                    error!("ContextManagerThread recv error: {}", err);
-                    break;
+                    return Err(ContextManagerThreadError::ThreadReceiveError(err));
```

Comment:
Now that you have the `From` implementation, you could change this to `match self.thread_receiver.recv()?` and remove the need for the `Ok`/`Err` items in the match arms.